buildpack/dockerfile.go

```diff
@@ -23,3 +30,171 @@ type ExtendArg struct {
 	Name  string `toml:"name"`
 	Value string `toml:"value"`
 }
+
+type InstructionCallback func(verb string, arg string, lineno int, state map[string]string) (string, error)
+
+func walkDockerInstructions(filename string, handleInstruction InstructionCallback) (string, error) {
```

Comment:
Did we look at using an existing library? I know kaniko has a [dockerfile package](https://github.com/GoogleContainerTools/kaniko/blob/239d16cd1c6301c997d511ea41b19cc0ab30621f/pkg/dockerfile/dockerfile.go#L212) that parses Dockerfiles or dockerfile commands?

It might not make sense given the requirements, but I figured I should ask.

Reply:
I didn't, but mostly under the belief it might be better to avoid dragging in a whole new dep for parsing dockerfiles.. that much said, @natalieparellano has pointed me at the buildkit parser that's in use by kaniko, and I'll have a look tomorrow at if I can't integrate that.. although the usage would be contained within the `walkDockerInstructions` method. 
Although it's a simple syntax, there's still decent benefits to be gained from using a lib with more testing etc =)

Reply:
I tried some experiments.. but I don't think we're going to use it.. (the kaniko/moby buildkit based dockerfile parsing library)

launcher with my parser - 2.2Mb
launcher with mobi buildkit parser fn welded to my visitor pattern - 5.7Mb

launcher with mobi buildkit parser fn with validation logic using mobi buildkit model - 5.7Mb
launcher using kaniko parser fn with validation logic using mobi buildkit model - 17Mb

(Kaniko parser fn was to try to avoid a bit of boilerplate around invoking the mobi buildkit parser.. definitely not worth the extra 11Mb!!)

The lifecycle build has a test to make sure the launcher comes in under 3Mb, and using the mobi buildkit parser/model is adding 3.5mb alone.

Reply:
Update: launcher shouldn't have had that package in it =) @natalieparellano found the problem.. looking good so far =)

Reply:
Resolved, moved to mobi / buildkit parsing =)